crates/angstrom-net/src/pool_manager.rs

```diff
@@ -83,6 +84,22 @@ impl OrderPoolHandle for PoolHandle {
         let _ = self.send(OrderCommand::CancelOrder(from, order_hash, tx));
         rx.map(|res| res.unwrap_or(false))
     }
+
+    fn estimate_gas(&self, order: AllOrders) -> impl Future<Output = Result<u64, String>> + Send {
+        let (tx, rx) = tokio::sync::oneshot::channel();
+        let _ = self.send(OrderCommand::EstimateGas(order, tx));
+        rx.map(|res| match res {
+            Ok(OrderValidationResults::Valid(o)) => Ok(o.priority_data.gas_units),
+            Ok(OrderValidationResults::Invalid(e)) => Err(format!("Invalid order: {}", e)),
+            Ok(OrderValidationResults::TransitionedToBlock) => {
+                Err("Order transitioned to block".to_string())
+            }
+            Err(e) => {
+                tracing::error!("could not estimate order gas {}", e);
+                Err(format!("Failed to estimate gas: {}", e))
+            }
+        })
+    }
```

Comment:
I think it would be more logical to just have this on the validator handle then the order pool